Sjq2t-17_authentication_be

DarkoVeljanoskiHtec opened the pull request:
Everything is set up for the authentication, I've added two new filters for the authentication and authorization and made changes to the web security class, register and login works.
The get users functionality is not implemented, I'll create new task on Jira for that.

DarkoVeljanoskiHtec committed:
Authentication init

DarkoVeljanoskiHtec committed:
Authentication init

DarkoVeljanoskiHtec committed:
added access and refresh token

DarkoVeljanoskiHtec committed:
added access and refresh token

DarkoVeljanoskiHtec committed:
Added Authorization filter

DarkoVeljanoskiHtec committed:
Added Authorization filter

DarkoVeljanoskiHtec committed:
Added Authorization filter, remove some lines

DarkoVeljanoskiHtec committed:
Added Authorization filter, remove some lines

DarkoVeljanoskiHtec committed:
Everything is working, except get user functionality

DarkoVeljanoskiHtec committed:
Everything is working, except get user functionality

DraganMitrasinovicHtec commented on backend/src/main/java/com/htecgroup/SkyNest/model/request/UserLoginRequest.java:
```diff
@@ -1,8 +1,12 @@
+import lombok.Setter;
 
 @Data
+@Setter
```
Doesn't Data already make all Getters and Setters?

DarkoVeljanoskiHtec replied:
I think you're right, I didn't know that.

DraganMitrasinovicHtec commented on backend/src/main/java/com/htecgroup/SkyNest/service/impl/UserServiceImpl.java:
```diff
@@ -51,4 +51,6 @@ public UserDto findUserByEmail(String email) {
 
     return modelMapper.map(userEntity, UserDto.class);
   }
+
```
Try to format or check diffs before pushing, so we don't get changes like this :) .

DarkoVeljanoskiHtec replied:
That's good suggestion, I'll check it from now.

DarkoVeljanoskiHtec committed:
little Refactoring

DarkoVeljanoskiHtec committed:
little Refactoring

ivanfhtec committed:
added registration db script, entities, dtos, and repositories (#12)

* added registration db script, entities, dtos, and repositories
switched repos to optional, where applicable
removed Utils as we'll be using UUIDs instead of generating our own strings

* added bean for ModelMapper and autowired it

* made flyway scripts for each statement

* switched to UserExceptions

* a couple of tests

* additional tests

* changed package name to lowercase

DarkoVeljanoskiHtec committed:
Merge issues - refactor

DarkoVeljanoskiHtec committed:
Merge issues - refactor

DraganMitrasinovicHtec commented on backend/src/main/java/com/htecgroup/SkyNest/filter/CustomAuthenticationFilter.java:
```diff
@@ -0,0 +1,65 @@
+        response.addHeader(SecurityConstants.HEADER_STRING, SecurityConstants.TOKEN_PREFIX + token);
+
+        response.addHeader("UserID", userDto.getId().toString());
+
```
Was it intended to have this line twice?

DarkoVeljanoskiHtec replied:
it was mistake, I changed it.

DarkoVeljanoskiHtec committed:
Merge issues - refactor

DarkoVeljanoskiHtec committed:
Merge issues - refactoring

DarkoVeljanoskiHtec committed:
Refactor class package

DarkoVeljanoskiHtec committed:
Delete backend/src/main/java/com/htecgroup/SkyNest directory

DarkoVeljanoskiHtec committed:
Refactor packages

DarkoVeljanoskiHtec committed:
rm double line

DarkoVeljanoskiHtec committed:
added new userExceptionType

DarkoVeljanoskiHtec committed:
rm unnecessary getters and setters

DarkoVeljanoskiHtec committed:
fix merge issues
